### PR TITLE
Initial logic to index Amundsen datasets according to v0 spec

### PR DESCRIPTION
**Pull request by munkyshi**
Implements a minio amundsen indexer according to the minio v0 schema specification.

Dependencies:
- Depends on the existence of a python spark docker image, which is currently only set up for build on 10.142.20.66.
- Depends on k8s cluster having a `default/spark` service account

### CODE REVIEW

**Comment by munkyshi** [comment]
I've successfully run a k8s job with an image built with `minioloader.Dockerfile`, so opening this up for review now.

File: databuilder/extractor/minio_extractor.py [hunk 1]
```diff
@@ -1,9 +1,14 @@
-from pyhocon import ConfigTree
-from typing import Any
+from typing import Any, Iterator
+
```
Comment (we-z) [inline databuilder/extractor/minio_extractor.py:2]: Extra whitespace between imports

File: databuilder/utils/spark_driver.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import socket
+from typing import Tuple
+
+import pyspark
+from pyspark import SparkContext
+from pyspark.sql.session import SparkSession
+
+from databuilder.utils.minio_conf import MinioConf
+
+
```
Comment (we-z) [inline databuilder/utils/spark_driver.py:11]: good addition

**Review by we-z — APPROVED** [review]
Looks great! ready to merge

File: databuilder/utils/spark_driver.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    conf.set("spark.hadoop.fs.s3a.path.style.access", True)
+    conf.set("spark.hadoop.fs.s3a.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem")
+    conf.set("spark.executor.instances", 1)
+    conf.set("spark.executor.memory", "4g")
+    conf.set("spark.driver.memory", "2g")
+    conf.set("spark.driver.hostname", driver_ip)
+    conf.set("spark.driver.host", driver_ip)
+    conf.set("spark.kubernetes.driver.container.image", spark_image)
+    conf.set("spark.kubernetes.executor.container.image", spark_image)
+    conf.set("spark.kubernetes.container.image.pullPolicy", "Never")
```
Comment (munkyshi) [inline databuilder/utils/spark_driver.py]: Should be `IfNotPresent`

File: databuilder/extractor/minio_extractor.py (hunk 1 above)
Comment (munkyshi) [reply databuilder/extractor/minio_extractor.py:2]: This is actually intentional -- I'm grouping my imports into three groups according to PEP-8:
```
Imports should be grouped in the following order:

Standard library imports.
Related third party imports.
Local application/library specific imports.

You should put a blank line between each group of imports.
```

https://www.python.org/dev/peps/pep-0008/

File: databuilder/utils/spark_driver.py (hunk 3 above)
Comment (munkyshi) [reply databuilder/utils/spark_driver.py]: Done.

**Comment by alexanderDoria** [comment]
looks good, will try out the SpecUtil functions today

**Review by alexanderDoria — APPROVED** [review]

**Pull request merged**